subprojects/pineapple/src/main/java/org/gradle/internal/file/FileType.java

```diff
@@ -19,5 +19,6 @@
 public enum FileType {
     RegularFile,
     Directory,
-    Missing
+    Missing,
+    Unavailable
```

Comment:
I was thinking that we were going to add a new `FileSystemLocationSnapshot` subtype, not just a file type.

Reply:
I'll investigate

Reply:
What should be the FileType of an `UnavailableFileSnapshot`, should I keep `FileType.Unavailable`

Reply:
I think so.

Reply:
Done